### ISSUE TITLE
The gradient of input has been computed twice

### ISSUE THREAD

**Issue by qwertymj** [jacobgil/pytorch-grad-cam#44]
Hi, when running your code, I find that the gradient of input is computed twice, 

In ` gradcam.py`
input.requires_grad_ is set True during preprocess at line 71

`
input = preprocessed_img.requires_grad_(True)
` 

after that, the GradCam model  call `one_hot.backward(retain_graph=True) `  at line 116, which compute the gradient of input first time (you can call `input.grad.cpu().data.numpy() `  to verify this), then the GuidedBackpropReLUModel compute the gradient of input again at line 198: `one_hot.backward(retain_graph=True) ` , which will add the gradient to the former one.

you should set the input.requires_grad_=False before compute GradCam and reset input.requires_grad_=True before compute GuidedBackprop

**Comment by SCoulY** [comment]
@qwertymj I agree with this. @jacobgil

**Comment by jacobgil** [comment]
Thanks for pointing this out!
